### PR TITLE
ddl: auto pause add index on TiKV disk full

### PR DESCRIPTION
**Pull request by expxiaoli**
### What problem does this PR solve?

Issue Number: ref #68354

Problem Summary:

DXF add-index can consume a large amount of TiKV disk space. If TiKV becomes disk-full during add-index backfill, the DDL job may fail instead of giving operators a chance to expand capacity and resume the same job.

### What changed and how does it work?

This PR auto-pauses DXF add-index DDL jobs when TiKV disk-full errors are detected.

- add typed TiKV disk-full error propagation for local ingest disk-space checks
- let DXF add-index tasks opt in to pause-on-TiKV-disk-full behavior
- move opted-in DXF tasks to `pausing`/`paused` when failed subtasks hit typed TiKV disk-full errors
- persist a DDL pause reason `tikv_disk_full` on the owning DDL job
- allow users to resume this specific system-paused DDL job after expanding TiKV capacity
- surface an actionable DDL error to the original `ALTER TABLE ... ADD INDEX` statement
- add targeted DDL, DXF scheduler/storage, ingest, and RealTiKV test coverage

Cherry-pick of commit `87c423368c296cf71e038b81d271364dbcebc7bc` onto `release-nextgen-202603-auto-pause`.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No need to test

Manual test:

```bash
GOPATH=$HOME/go GOSUMDB=sum.golang.org make bazel_prepare
```

Original commit validation included targeted failpoint tests for DDL, DXF scheduler/storage, ingest, and compile-level RealTiKV validation. The RealTiKV end-to-end case was added but local execution was blocked by a TiUP playground/PD RPC compatibility issue.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
Auto-pause DXF add-index DDL jobs when TiKV becomes disk-full, so operators can expand capacity and resume the paused job.
```

## Summary by CodeRabbit

* **New Features**
  * TiDB DDL jobs now auto-pause on TiKV “disk full” (including distributed ADD-INDEX and multi-schema changes), with a dedicated resume/pause reason.
  * Scheduler and task execution support an opt-in “pause on KV disk full” mode.
* **Bug Fixes**
  * Resuming paused DDL jobs now clears pause reason and job error (and enforces operator matching, with an end-user recovery exception).
  * System-wide and upgrade-finish resume logic skips KV-disk-full system-paused jobs; multi-schema and executor handling were aligned.
* **Tests**
  * Added/updated unit and real TiKV tests covering scheduler pausing, DDL resume edge cases, and distributed add-index auto-pause.

### CODE REVIEW

File: pkg/dxf/framework/storage/task_table.go [hunk 1]
```diff
@@ -763,6 +763,38 @@ func (mgr *TaskManager) GetSubtaskCntGroupByStates(ctx context.Context, taskID i
...
+		cntByStates[state]++
+		if state != proto.SubtaskStateFailed && state != proto.SubtaskStateCanceled {
+			continue
+		}
+		subTaskErr, err := unmarshalSubtaskError(row.GetBytes(1), row.IsNull(1))
+		if err != nil {
+			return nil, nil, err
+		}
+		subTaskErrors = append(subTaskErrors, subTaskErr)
+	}
```
[the remark this answers is not in the document]
Comment (expxiaoli) [reply pkg/dxf/framework/storage/task_table.go:796]: Thanks for the review. I think we should keep nil entries here intentionally: the slice preserves one error slot per failed/canceled subtask. shouldPauseOnKVDiskFull relies on that count matching to avoid auto-pausing when any failed subtask has no classifiable KVDiskFull error.

**Review by wjhuang2016 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM
